File: exercise_7/play-backend/app/models/UserAddress.scala
package models

import play.api.libs.json._

case class UserAddress(id: Long = 0, userId: Long, firstname: String, lastname: String, address: String,
                       zipcode: String, city: String, country: String)

object UserAddress {
  implicit val userAddressFormat: OFormat[UserAddress] = Json.using[Json.WithDefaultValues].format[UserAddress]
}

File: exercise_6/react-frontend/src/services/userAddress.ts
import axios from 'axios';

const HOST = 'http://localhost:9000'

export const addUserAddress = async (userId: number, firstname: string, lastname: string, address: string, zipcode: string, city: string, country: string) => {
	return axios.post(`${HOST}/api/user-address/create`, {
		userId: userId,
		firstname: firstname,
		lastname: lastname,
		address: address,
		zipcode: zipcode,
		city: city,
		country: country,
	})
}

export const listUserAddresses = async (userId: number) => {
	return axios.get(`${HOST}/api/user-address/list-by-user/${userId}`)
}
